opendevin/controller/agent_controller.py

```diff
@@ -152,10 +154,13 @@ async def _run(self):
             try:
                 finished = await self.step(i)
                 if finished:
+                    self.finish_state = self.get_state()
                     await self.set_agent_state_to(AgentState.FINISHED)
                     break
-            except Exception:
+            except Exception as e:
                 logger.error('Error in loop', exc_info=True)
+                self.finish_state = self.get_state()
+                self.finish_state.error = str(e)
```

Comment:
Want to get clear about our logic, do we need the state after every step or only need the final one?

Reply:
We need one after the final step -- in this case when error occurs, it will `break` the loop, hence end the `._run` function. But i guess you caught a bug - we shouldn't re-get the state at the end of the lopp

Reply:
should be hopefully fixed in the new commit